src/os/net/baseservermodifier.js

```diff
@@ -1,68 +1,15 @@
-goog.provide('os.net.BaseServerExpression');
 goog.provide('os.net.BaseServerModifier');
 goog.require('goog.Uri');
 goog.require('os.net.URLModifier');
 
 
-
-/**
- * @extends {os.net.URLModifier}
- * @constructor
- */
-os.net.BaseServerModifier = function() {
-  os.net.BaseServerModifier.base(this, 'constructor', 'baseServer');
-};
-goog.inherits(os.net.BaseServerModifier, os.net.URLModifier);
-
-
-/**
- * @type {!Array<{search: RegExp, replace: string}>}
- * @private
- */
-os.net.BaseServerModifier.replace_ = [];
-
-
-/**
- * The expression to remove https and trailing slashes
- * @type {RegExp}
- */
-os.net.BaseServerExpression = /(https:\/\/)?(.*[^\/])\/?/;
-
-
 /**
  * @param {string} server
  */
 os.net.BaseServerModifier.configure = function(server) {
-  os.net.BaseServerModifier.replace_.length = 0;
-
   if (server) {
-    os.net.BaseServerModifier.replace_.push({
-      search: new RegExp(os.settings.get('baseServerRegex', '^\\/(.*)$')),
-      replace: server.replace(/\/$/, '') + '/$1'
-    });
+    var config = {};
+    config[os.settings.get('baseServerRegex', '^\\/([^\\/].*)$')] = server.replace(/\/$/, '') + '/$1';
```

Comment:
The regex change here fixes scheme-relative URLs (e.g. `//some.domain.com/...`)

Reply:
Make sure all stakeholders and downstream projects work with this change